### PR TITLE
Fixes #922. Remove dependency on Jersey for SSLContext configuration

### LINKED ISSUE #922: Unable to configure TLS on NettyDockerCmdExecFactory without Jersey
I was hoping that I'd be able to cut Jersey out of my dependency tree and use Netty via the `NettyDockerCmdExecFactory`.

With TLS configured I hit the following issue. Do we really need Jersey to configure SSL, even when Netty is used as the client library?

```
Caused by: java.lang.NoClassDefFoundError: org/glassfish/jersey/SslConfigurator
	at com.github.dockerjava.core.LocalDirectorySSLConfig.getSSLContext(LocalDirectorySSLConfig.java:56)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory$InetSocketInitializer.initSsl(NettyDockerCmdExecFactory.java:340)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory$InetSocketInitializer.connect(NettyDockerCmdExecFactory.java:322)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory.connect(NettyDockerCmdExecFactory.java:220)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory.connect(NettyDockerCmdExecFactory.java:213)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory.access$000(NettyDockerCmdExecFactory.java:157)
	at com.github.dockerjava.netty.NettyDockerCmdExecFactory$1.getChannel(NettyDockerCmdExecFactory.java:183)
	at com.github.dockerjava.netty.InvocationBuilder.getChannel(InvocationBuilder.java:213)
	at com.github.dockerjava.netty.InvocationBuilder.get(InvocationBuilder.java:196)
	at com.github.dockerjava.netty.InvocationBuilder.get(InvocationBuilder.java:187)
	at com.github.dockerjava.netty.exec.ListImagesCmdExec.execute(ListImagesCmdExec.java:41)
	at com.github.dockerjava.netty.exec.ListImagesCmdExec.execute(ListImagesCmdExec.java:16)
	at com.github.dockerjava.netty.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
	at com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
```

### PR DESCRIPTION
**Pull request by jamesnetherton**
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/936)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/core/LocalDirectorySSLConfig.java [hunk 1]
```diff
@@ -53,13 +56,20 @@ public SSLContext getSSLContext() {
...
-                sslConfig.keyStorePassword("docker");
-                sslConfig.trustStore(CertificateUtils.createTrustStore(capem));
+                String kmfAlgorithm = AccessController.doPrivileged(getSystemProperty("ssl.keyManagerFactory.algorithm",
+                    KeyManagerFactory.getDefaultAlgorithm()));
+                KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(kmfAlgorithm);
+                keyManagerFactory.init(CertificateUtils.createKeyStore(keypem, certpem), "docker".toCharArray());
 
-                return sslConfig.createSSLContext();
+                String tmfAlgorithm = AccessController.doPrivileged(getSystemProperty("ssl.trustManagerFactory.algorithm",
+                    TrustManagerFactory.getDefaultAlgorithm()));
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/LocalDirectorySSLConfig.java]: why it needed?

File: src/main/java/com/github/dockerjava/core/LocalDirectorySSLConfig.java [hunk 2]
```diff
@@ -53,13 +56,20 @@ public SSLContext getSSLContext() {
                 String certpem = new String(Files.readAllBytes(Paths.get(certPemPath)));
                 String capem = new String(Files.readAllBytes(Paths.get(caPemPath)));
 
-                SslConfigurator sslConfig = SslConfigurator.newInstance(true);
-                sslConfig.securityProtocol("TLSv1.2");
-                sslConfig.keyStore(CertificateUtils.createKeyStore(keypem, certpem));
-                sslConfig.keyStorePassword("docker");
-                sslConfig.trustStore(CertificateUtils.createTrustStore(capem));
+                String kmfAlgorithm = AccessController.doPrivileged(getSystemProperty("ssl.keyManagerFactory.algorithm",
+                    KeyManagerFactory.getDefaultAlgorithm()));
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/LocalDirectorySSLConfig.java]: and this

File: src/main/java/com/github/dockerjava/core/LocalDirectorySSLConfig.java (hunk 1 above)
Comment (jamesnetherton) [reply src/main/java/com/github/dockerjava/core/LocalDirectorySSLConfig.java]: @KostyaSha any thoughts on my last comment?

**Comment by KostyaSha** [comment]
CC @bsideup

**Comment by bsideup** [comment]
@jamesnetherton thanks! Very valuable for http://github.com/testcontainers/testcontainers-java project!

**Pull request merged**